### PR TITLE
Omit `changes` query parameter in `CreateImageOptions` if `Vec` is empty

### LINKED ISSUE #352: `DockerResponseServerError { status_code: 400, message: "file with no instructions" }`
The following function for creating a WASM image in the local registry works in `bollard 0.14.0` but fails with `DockerResponseServerError { status_code: 400, message: "file with no instructions" }` in `bollard 0.15.0`.

I have bisected the commits, and it appears that it was merge request [https://github.com/fussybeaver/bollard/pull/335](https://github.com/fussybeaver/bollard/pull/335) that caused the issue, but I can't find out how sending an empty `changes` field causes the error.

```
/// Creates an image in the local registry from the provided WASM module bytes
pub async fn create_image_from_slice(
    image_name: &str,
    tag: &str,
    module: &[u8],
) -> eyre::Result<()> {
    let options = CreateImageOptions {
        from_src: "-",    // from_src must be "-" when sending the archive in the request body
        repo: image_name, // The name of the image in the docker daemon.
        tag,              // The tag of this particular image.
        ..Default::default()
    };

    // Create a Body from the byte slice
    let req_body = hyper::body::Body::from({
        let mut buffer = Vec::new();

        {
            let mut builder = tar::Builder::new(&mut buffer);
            let mut header = tar::Header::new_gnu();
            header.set_path("entrypoint.wasm")?;
            header.set_size(module.len() as u64);
            header.set_cksum();

            builder.append_data(&mut header, "entrypoint.wasm", module)?;
            builder.finish()?;
        }

        buffer
    });

    // Call Docker::create_image with the options and the body
    Docker::connect_with_local_defaults()?
        .create_image(Some(options), Some(req_body), None)
        .collect::<Vec<_>>()
        .await
        .into_iter()
        .collect::<Result<Vec<_>, bollard::errors::Error>>()
        .unwrap();

    Ok(())
}
```

### PR DESCRIPTION
**Pull request by fmckeogh**
Closes #352

### CODE REVIEW

**Comment by fussybeaver** [comment]
I'm not on my desktop at the moment, but looks good... Would also be good to get a test case around the wasm use case that you mentioned.

**Comment by fmckeogh** [comment]
> I'm not on my desktop at the moment, but looks good... Would also be good to get a test case around the wasm use case that you mentioned.

Sure thing, I'll write one now (making sure it fails without the fix)

**Comment by fmckeogh** [comment]
Done

File: tests/image_test.rs [hunk 1]
```diff
@@ -27,6 +27,41 @@ async fn create_image_test(docker: Docker) -> Result<(), Error> {
...
+    });
+
+    docker
+        .create_image(Some(options), Some(req_body), None)
+        .collect::<Vec<_>>()
+        .await
+        .into_iter()
+        .collect::<Result<Vec<_>, bollard::errors::Error>>()
+        .unwrap();
+
```
Comment (fussybeaver) [inline tests/image_test.rs:61]: Could we assert that the image was built by using `inspect_image`?

**Comment by fussybeaver** [comment]
Thanks, looks good, happy to merge if we can get one assert in there

File: tests/image_test.rs (hunk 1 above)
Comment (fmckeogh) [reply tests/image_test.rs:63]: Done

**Pull request merged**